### PR TITLE
Doc 8288 - Implement "Category facet" feature

### PR DESCRIPTION
**Pull request by dlutzCoveo**
(updated) link:
https://codesandbox.io/s/category-facet-breadcrumbs-forked-tmh9b

Sorry - The copy/pasta approach was more efficient by page, but reflects oddly in the way files have changed here in GiuHub.

### CODE REVIEW

File: src/Components/CategoryFacet.tsx [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+const hoveredStyle = {
+  cursor: "pointer",
+  color: "primary"
+};
+
+interface ICategoryFacetProps {
+  title: string;
+  field: string;
+  // basePath: [];
```
Comment (kaitlynkatz) [inline src/Components/CategoryFacet.tsx:23]: One comment you forgot :)

**Review by kaitlynkatz — APPROVED** [review]
Looks great! One thing: I know it's hard to keep track of what files changes because we lost our code sandbox-github integration, but one important change you forgot to copy over is actually adding the category facet on the page :) that goes in App.tsx
But the code sandbox is really looking great, I like the alignment solution you came up with! Nice work :)

File: src/Components/CategoryFacet.tsx [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    this.setState(this.headlessCategoryFacet.state);
+  }
+
+  //sends left position offset value to getValues <Box> depending on state
+  parentAlign() {
+    let align = "0%";
+    if (this.state.hasActiveValues) {
+      align = "20%";
+    }
+    return align;
```
[the remark this answers is not in the document]
Comment (dlutzCoveo) [reply src/Components/CategoryFacet.tsx:64]: Ah, that's SO much better. Ternary for the win.

File: src/Components/CategoryFacet.tsx (hunk 1 above)
Comment (dlutzCoveo) [reply src/Components/CategoryFacet.tsx:23]: OK, I fixed those two issues and have synced. Thank you!

**Review by jpmarceau — COMMENTED** [review]
Very nice!
One thing I think we should also adjust is the breadcrumbs, i.e., selecting category facet values should add corresponding breadcrumb values.

**Comment by fbeaudoincoveo** [comment]
As @jpmarceau mentioned, we should support breadcrumbs. Here's are a images showing how they work for hierarchical facets in JS UI:

Screen Shot 2021-02-12 at 07 05 20
Screen Shot 2021-02-12 at 07 05 25

Another thing I think we should add is the show more / show less button:

Screen Shot 2021-02-12 at 07 08 19
Screen Shot 2021-02-12 at 07 08 26

**Review by fbeaudoincoveo — COMMENTED** [review]
Great job :)

File: src/App.tsx [hunk 3]
```diff
@@ -46,8 +54,13 @@ export default class App extends React.Component {
             <FacetBreadcrumbs />
             <Grid container>
               <Grid item xs={4}>
-                <Facet title="Source" field="source" />
+                <Facet title="Source" field="atlgeographicalhierarchy" />
```
Comment (jpmarceau) [inline src/App.tsx:57]: I guess you'll set that one back to `source` before merging 🤔

File: src/Components/CategoryFacet.tsx [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+          <FormLabel component="legend" color="primary">
+            {this.props.title}
+          </FormLabel>
+        </Box>
+        {this.getClearButton()}
+        {this.getParents()}
+        {this.getValues()}
+        {this.state.canShowMoreValues && this.getShowMore()}
+        {this.state.canShowLessValues && this.getShowLess()}
+      </Box>
```
Comment (jpmarceau) [inline src/Components/CategoryFacet.tsx:214]: Do you want to include getCategoryFacetSearch() as well?

**Review by jpmarceau — COMMENTED** [review]
Very good work Danny! Nice to see you ramp up :)

Besides the two comments I just left, here's one extra thing:
You should probably sync your branch with the latest 0.1.0 (which is where you'll want to merge as well rather than main, btw), because I merged some changes in there this morning. Because of the sketchy copy-paste we're doing right now for having lost our github-codesandbox integration, it would be good to create a new sandbox from your updated branch, to validate that everything is OK :)

Don't hesitate to ping me if you'd like to talk about this 👍

**Comment by dlutzCoveo** [comment]
Closing - this has been transferred to **DOC 8288 - V2** Which is a branch from 0.1.0 rather than main.

**Pull request closed without merging**